### ISSUE TITLE
‘Typeable’ changes prevent compilation

### ISSUE THREAD

**Issue by dbaynard** [mikeizbicki/vector-heterogenous#4]
With ghc 7.8.2, vector 0.10.9.2, and base 4.7.0.0, vector-heterogeneous does not compile.

A quick search suggests this version of base breaks the functionality of Typeable. https://ghc.haskell.org/trac/ghc/wiki/GhcKinds/PolyTypeable ([via](https://www.haskell.org/ghc/docs/7.8.2/html/libraries/base-4.7.0.0/Data-Typeable.html)) gives two solutions — one, to change to ‘OldTypeable’ in the short term, and the other to change the code to be compatible with the new version..

```
[1 of 3] Compiling Data.Vector.Heterogenous.HList ( src/Data/Vector/Heterogenous/HList.hs, dist/build/Data/Vector/Heterogenous/HList.o )

src/Data/Vector/Heterogenous/HList.hs:113:5:
    ‘typeOf’ is not a (visible) method of class ‘Typeable’

src/Data/Vector/Heterogenous/HList.hs:270:1:
    Kind variable also used as type variable: ‘a’
    In the declaration for type family ‘$’
```

I’m using an arch linux PKGBUILD from cabal2arch, but straight-up cabal gives the same problem.

I’d happily give it a go when I get some free time, but that may be in a few weeks. I’m quite curious to try some of the code from your blog.

p.s. I just saw ‘- Typeable is scary and I don't understand what's going on.  Hopefully this is correct :)’. Please accept my apologies.

**Comment by mikeizbicki** [comment]
Hmm... it looks like either of those fixes causes breakage with GHC 7.6, right?

That's okay by me.  It probably shouldn't be too hard to upgrade to the new `Typeable`, which would be the right longterm solution.

**Comment by tcfuji** [comment]
I'm getting the same error now with GHC 7.8.3 on OS X 10.9. Should I just downgrade back to 7.6? If not, how can this be fixed (again, for a Haskell beginner)?

**Comment by mikeizbicki** [comment]
I've fixed the errors and uploaded a new version (0.2.0) to hackage.
